cupy/core/core.pyx

```diff
@@ -3121,20 +3128,23 @@ cpdef _prepare_mask_indexing_single(ndarray a, ndarray mask, Py_ssize_t axis):
     masked_shape = lshape + (n_true,) + rshape
 
     # When mask covers the entire array, broadcasting is not necessary.
-    if mask.ndim == a.ndim and axis == 0:
+    if mask._shape.size() == a._shape.size() and axis == 0:
```

Comment:
`.ndim` and `._shape.size()` are mixed in the function.  Could you explain your policy on the choices?